vformer/models/classification/visformer.py

```diff
@@ -0,0 +1,182 @@
+import einops
+import torch.nn as nn
+
+from vformer.attention import VanillaSelfAttention
+
+from ...utils import MODEL_REGISTRY
+
+
+# need to add visformerV2_ti
+# editted number of need to change
+# run one epoch to check if the model is working
+class Conv_Block(nn.Module):
+    def __init__(self, in_channels, group=8, activation=nn.GELU, drop=0.0):
+        super(Conv_Block, self).__init__()
+        self.norm1 = nn.BatchNorm2d(in_channels)
+        self.conv1 = nn.Conv2d(in_channels, in_channels * 2, kernel_size=1, bias=False)
+        self.act1 = activation()
+        self.conv2 = nn.Conv2d(
+            in_channels * 2,
+            in_channels * 2,
+            kernel_size=3,
+            padding=1,
+            groups=group,
+            bias=False,
+        )
+        self.act2 = activation()
+        self.conv3 = nn.Conv2d(in_channels * 2, in_channels, kernel_size=1, bias=False)
+        self.drop = nn.Dropout(drop)
+
+    def forward(self, x):
+        xt = x
+        xt = self.norm1(xt)
+        x = self.conv1(x)
+        x = self.act1(x)
+        x = self.drop(x)
+        x = self.conv2(x)
+        x = self.act2(x)
+        x = self.conv3(x)
+        x = xt + self.drop(x)
+        del xt
+        return x
+
+
+class Attention_Block(nn.Module):
+    def __init__(self, channels, num_heads=8, drop=0.0):
+        super().__init__()
+        self.conv1 = nn.Conv2d(channels, channels * 4, kernel_size=1, bias=False)
+        self.conv2 = nn.Conv2d(channels * 4, channels, kernel_size=1, bias=False)
+        self.attn = VanillaSelfAttention(
+            channels, num_heads=num_heads, head_dim=channels // num_heads
+        )
+        self.norm1 = nn.BatchNorm2d(channels)
+        self.norm2 = nn.BatchNorm2d(channels)
+        self.drop = nn.Dropout(drop)
+
+    def forward(self, x):
+        B, C, H, W = x.shape
+        xt = einops.rearrange(x, "b c h w -> b (h w) c")
+        x = self.norm1(x)
+        x = einops.rearrange(x, "b c h w -> b (h w) c")
+        x = self.attn(x)
+        x = xt + x
+        x = einops.rearrange(x, "b (h w) c -> b c h w", h=H, w=W)
+        xt = x
+        x = self.norm2(x)
+        x = self.conv1(x)
+        x = self.conv2(x)
+        x = self.drop(x)
+        x = xt + x
+        del xt
+        return x
+
+
+@MODEL_REGISTRY.register()
+class Visformer(nn.Module):
+    def __init__(
+        self,
+        image_size,
+        n_classes,
+        depth: list,
+        config: str,
+        channel_config: list,
+        num_heads=8,
+        drop=0.0,
```

Comment:
@abhi-glitchhg had made some changes where he had standardized the names of parameters like dropout across all files. Could you check if this is consistent with those names?

Reply:
alright lemme see